book/src/customexample.md

```diff
@@ -0,0 +1,79 @@
+
+# Ethdos custom predicate, using binary AND and OR: example of a direct definition and a recursive batch
+
+## Direct definition
+```
+eth_friend(src_or, src_key, dst_or, dst_key) = and<
+    // there is an attestation pod that's a SIGNATURE POD
+    ValueOf(attestation_pod, "type", SIGNATURE)     
+    
+    // the attestation pod is signed by (src_or, src_key)
+    Equal((attestation_pod, "signer"), (src_or, src_key))  
+
+    // that same attestation pod has an "attestation"
+    Equal((attestation_pod, "attestation"), (dst_or, dst_key))
+>
+
+On the backend, this is converted to a rule where all the arguments are either wildcards or literals:
+eth_friend(*1, *2, *3, *4) = and<
```

Comment:
This is a very interesting example to discuss the topic of arities in custom predicates.
This custom predicate has:
- The predicate has 4 arguments (that's what I was calling public  in https://github.com/0xPARC/pod2/issues/54)
- The predicate has 5 wildcards
- The predicate has 1 wildcard that doesn't come from a predicate argument (that's what I was calling private in https://github.com/0xPARC/pod2/issues/54)